src/h5web/vis-packs/core/hooks.ts

```diff
@@ -34,11 +33,7 @@ export function useDatasetValue(
     return undefined;
   }
 
-  // If no dim mapping is provided or dim mapping has no slicing dimension, fetch entire dataset
-  if (!dimMapping || !dimMapping.some(isNumber)) {
-    return valuesStore.get({ path: dataset.path });
-  }
```

Comment:
This logic is now inside `getSliceSelection` which is called when performing the eviction in the Heatmap vis error boundary, so that `evict` is always called with the same object as `get`. The bug was that, when `dimMapping` had no slicing dimension, `evict` was called with an empty string `selection` (instead of just `undefined` like above)